### PR TITLE
[3.8.4] optimize SafeArea & fix ios onOrientationChanged bug

### PR DESCRIPTION
**Pull request by yanjifa**
Re: #

### Changelog

* optimize SafeArea
* Fix the issue where SafeArea doesn't change during the issuance of the iOS onOrientationChanged notification

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/platform/ios/AppDelegateBridge.mm [hunk 1]
```diff
@@ -70,40 +70,21 @@ - (float)getPixelRatio {
     return (float)screenIntf->getDevicePixelRatio();
 }
 
-- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
-    cc::IScreen::Orientation orientation;
```
Comment (yanjifa) [inline native/cocos/platform/ios/AppDelegateBridge.mm:74]: 这个变量后面的逻辑没有使用， 删除了

File: native/cocos/platform/ios/AppDelegateBridge.mm [hunk 2]
```diff
@@ -70,40 +70,21 @@ - (float)getPixelRatio {
...
-
     float pixelRatio = screenIntf->getDevicePixelRatio();
     cc::WindowEvent resizeEv;
     resizeEv.windowId = 1;
     resizeEv.type = cc::WindowEvent::Type::RESIZED;
     resizeEv.width = size.width * pixelRatio;
     resizeEv.height = size.height * pixelRatio;
     cc::events::WindowEvent::broadcast(resizeEv);
-}
+    [coordinator animateAlongsideTransition:nil completion:^(id<UIViewControllerTransitionCoordinatorContext> context) {
```
Comment (yanjifa) [inline native/cocos/platform/ios/AppDelegateBridge.mm:84]: 要等旋转屏幕动画完成后，获取 SafeArea 才是正确的

File: pal/screen-adapter/minigame/screen-adapter.ts [hunk 3]
```diff
@@ -125,23 +125,10 @@ class ScreenAdapter extends EventTarget {
         // NOTE: safe area info on vivo platform is in physical pixel.
         // No need to multiply with DPR.
         const dpr = VIVO ? 1 : this.devicePixelRatio;
-        let topEdge = minigameSafeArea.top * dpr;
-        let bottomEdge = windowSize.height - minigameSafeArea.bottom * dpr;
-        let leftEdge = minigameSafeArea.left * dpr;
-        let rightEdge = windowSize.width - minigameSafeArea.right * dpr;
-        const orientation = this.orientation;
-        // Make it symmetrical.
```
Comment (yanjifa) [inline pal/screen-adapter/minigame/screen-adapter.ts:133]: 对称操作放到 sys.ts 里面做

File: cocos/core/platform/sys.ts [hunk 4]
```diff
@@ -361,11 +362,25 @@ export const sys = {
      * 返回基于游戏视图坐标系的手机屏幕安全区域（设计分辨率为单位），如果不是异形屏将默认返回一个和 visibleSize 一样大的 Rect。
      * 目前支持安卓、iOS 原生平台和微信、字节小游戏平台。
      * @method getSafeAreaRect
+     * @param [symmetric=true] @zh 基于屏幕对称的 Rect。 @en Rect that is symmetric based on the screen.
      * @return {Rect}
      */
-    getSafeAreaRect (): Rect {
+    getSafeAreaRect (symmetric: boolean = true): Rect {
```
Comment (yanjifa) [inline cocos/core/platform/sys.ts:368]: symmetric 默认 true，保持和之前的逻辑一致

**Comment by yanjifa** [comment]
iOS 竖屏转横屏会出现异常


SafeArea 支持选择是否对称， 浅紫色为选择对称的情况，浅蓝色为选择不对称的情况
截屏2024-07-12 02 49 36

**Comment by yanjifa** [comment]
https://github.com/cocos/cocos-engine/issues/17356

**Comment by yanjifa** [comment]
@minggo Please assign someone to review this PR.

**Comment by minggo** [comment]
@yanjifa thanks.
@qiuguohua please take a look.

**Comment by yanjifa** [comment]
@qiuguohua please take a look

File: native/cocos/platform/ios/AppDelegateBridge.mm (hunk 2 above)
Comment (qiuguohua) [reply native/cocos/platform/ios/AppDelegateBridge.mm:84]: 这里我用模拟器测试了下，从竖屏到横屏的旋转，没有进入。

Comment (yanjifa) [reply native/cocos/platform/ios/AppDelegateBridge.mm:84]: 我测试的时候能旋转，构建时候有选横屏和竖屏吗

Comment (yanjifa) [reply native/cocos/platform/ios/AppDelegateBridge.mm:84]: test

**Review by qiuguohua — APPROVED** [review]

File: native/cocos/platform/ios/AppDelegateBridge.mm (hunk 2 above)
Comment (qiuguohua) [reply native/cocos/platform/ios/AppDelegateBridge.mm:84]: 好的，我勾选错了。可以了

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
@qiuguohua  please update the release note

**Comment by qiuguohua** [comment]
OK

**Pull request merged**